server/src/main/java/org/elasticsearch/cluster/metadata/ColumnPositionResolver.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Licensed to Crate.io GmbH ("Crate") under one or more contributor
+ * license agreements.  See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.  Crate licenses
+ * this file to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial agreement.
+ */
+
+package org.elasticsearch.cluster.metadata;
+
+import javax.annotation.Nonnull;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Map;
+import java.util.TreeMap;
+import java.util.function.BiConsumer;
+
+public class ColumnPositionResolver<T> {
+    // Depths of the columns are used as keys such that deeper columns take higher column positions. (parent's position < children's positions)
+    private final Map<Integer, List<Column<T>>> columnsToReposition = new TreeMap<>(Comparator.naturalOrder());
+    private int maxColumnPosition = 0;
+
+    public static <T> ColumnPositionResolver<T> resolve(@Nonnull ColumnPositionResolver<T> toResolve) {
```

Comment:
I got the impression that this method does two things?

Creating a new instance which is only used at one call-site, and processing all columns added so far. Could this be split into up?

E.g have a instance method to process the columns, and the merge could be done directly in the `RootObjectMapper`

Reply:
Thank you, this is also handled by [5a4edc6](https://github.com/crate/crate/pull/12754/commits/5a4edc6e7cee604595ecd4ba8956881d67aeee86) and [357a286](https://github.com/crate/crate/pull/12754/commits/357a28657b60712a8eb06ab80622753a8dc88260)